packages/core/src/browser/label-provider.ts

```diff
@@ -82,12 +82,51 @@ export interface LabelProviderContribution {
      */
     affects?(element: object, event: DidChangeLabelEvent): boolean;
 
+    getUriLabel?(element: object): string;
+
+    registerFormatter?(formatter: ResourceLabelFormatter): Disposable;
```

Comment:
it should belong only to `DefaultUriLabelProviderContribution`

Reply:
We need the `registerFormatter()` function to pass the formatter to label provider.

Reply:
My concern is that only `DefaultUriLabelProviderContribution` is interested into it right now. If any another contribution won't to make use of it, they will need to duplicate much of `DefaultUriLabelProviderContribution`. 

If we want to allow other contribution to reuse it, it would be better to have something like `ResourceLabelService` which collects formatters and expose `formatUri` and change event for `DefaultUriLabelProviderContribution` and other contributions.

Reply:
Applied the second option.